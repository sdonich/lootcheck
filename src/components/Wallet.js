import React from 'react';
import { connect } from 'react-redux';

export class Wallet extends React.Component {
  render() {
    return (
      <div>
        <h3 className="balance">
          Wallet balance: {this.props.balance}
        </h3>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    balance: state
  }
}

export default connect(mapStateToProps)(Wallet);