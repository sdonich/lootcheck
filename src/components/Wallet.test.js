import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { Wallet } from './Wallet';

describe('Wallet', () => {
  const props = { balance: 20 };
  const wallet = shallow(<Wallet {...props} />);

  test('renders properly', () => {
    const tree = renderer.create(<Wallet />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('displays the balance from props', () => {
    expect(wallet.find('.balance').text()).toBe('Wallet balance: 20');
  });
});