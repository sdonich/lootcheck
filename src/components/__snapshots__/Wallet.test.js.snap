// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders properly 1`] = `
<div>
  <h3
    className="balance"
  >
    Wallet balance: 
  </h3>
</div>
`;
